hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/bootstrap/translator/DecodedBootstrapPartitionPathTranslator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client.bootstrap.translator;
+
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.common.util.PartitionPathEncodeUtils;
+
+/**
+ * URI decodes the partition path
+ *
+ */
+public class DecodedBootstrapPartitionPathTranslator extends BootstrapPartitionPathTranslator {
+  public DecodedBootstrapPartitionPathTranslator(TypedProperties properties) {
+    super(properties);
```

Comment:
Do we need `properties`?

Reply:
No. I originally limited it to only simple keygen, but removed that restriction